src/validator.ts

```diff
@@ -97,12 +97,46 @@ const typeErrors = (
   messageParts: Array<string>
 ) => {
   const typeErrorsList: Array<string> = [];
+  const anyOfConstPropErrors: Array<ErrorObject> = errors.filter(
+    (error) =>
+      error.schemaPath.indexOf('anyOf') > -1 &&
+      error.keyword === 'const' &&
+      error.instancePath.length > 0
+  );
   let errorCount = 0;
 
+  // handle possible literal types first
+  const propErrorGroups: { [key: string]: Array<ErrorObject> } = {};
+  if (anyOfConstPropErrors.length > 0) {
+    for (const error of anyOfConstPropErrors) {
+      const constValue = error.instancePath.slice(1);
+
+      if (propErrorGroups[constValue]) {
+        propErrorGroups[constValue].push(error);
+      } else {
+        propErrorGroups[constValue] = [error];
+      }
+    }
+    const properties = Object.keys(propErrorGroups);
+
+    properties.forEach((property) => {
+      const constValueErrors = propErrorGroups[property];
+
+      typeErrorsList.push(
+        `property '${property}' is a constant which must be equal to one of: ` +
```

Comment:
I think we could make this slightly more concise.

```suggestion
        `property '${property}' must be equal to one of: ` +
```

Reply:
Good idea, thanks!